packages/ember-environment/lib/global.js

```diff
@@ -14,5 +14,4 @@ function checkElementIdShadowing(value) {
 export default checkGlobal(checkElementIdShadowing(typeof global === 'object' && global)) ||
                checkGlobal(typeof self === 'object' && self) ||
                checkGlobal(typeof window === 'object' && window) ||
-               mainContext || // set before strict mode in Ember loader/wrapper
```

Comment:
And also add a `// eslint-disable-next-line no-unused-vars` line before the mainContext definition in packages/loader/lib/index.js (even though no-unused-vars is globally off for now)

Still, i'm a bit puzzled by the global.js code... If mainContext is defined nowhere, the code will still fail. And if it is, the next line `new Function('return this')()` will never run. 

Would it make sense to replace `mainContext` with `(typeof mainContext === 'object' && mainContext)` ?